File: exo/com/exo/gui/client/GuiAssembler.java
package com.exo.gui.client;

import net.minecraft.client.gui.inventory.GuiContainer;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.util.ResourceLocation;

import org.lwjgl.opengl.GL11;

import com.exo.gui.server.ContainerAssembler;
import com.exo.tile.TileAssembler;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public final class GuiAssembler extends GuiContainer{
	private final ResourceLocation TEXTURE = new ResourceLocation("exo", "textures/gui/guiAssembler.png");
	private final TileAssembler tile;
	
	public GuiAssembler(InventoryPlayer player, TileAssembler tile){
		super(new ContainerAssembler(player, tile));
		this.tile = tile;
	}

	@Override
	protected void drawGuiContainerBackgroundLayer(float f, int i, int j) {
		GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
		this.mc.getTextureManager().bindTexture(this.TEXTURE);
		int k = (this.width - this.xSize) / 2;
		int l = (this.height - this.ySize) / 2;
		this.drawTexturedModalRect(k, l, 0, 0, this.xSize, this.ySize);
	}
	
	@Override
	public void drawGuiContainerForegroundLayer(int x, int y){
		
	}
}

File: exo/com/exo/client/gui/GuiAssembler.scala
package com.exo.client.gui

import com.exo.server.gui.ContainerAssembler
import com.exo.tile.TileAssembler
import cpw.mods.fml.relauncher.SideOnly
import cpw.mods.fml.relauncher.Side
import net.minecraft.client.gui.inventory.GuiContainer
import net.minecraft.entity.player.InventoryPlayer
import net.minecraft.util.ResourceLocation
import org.lwjgl.opengl.GL11

@SideOnly(Side.CLIENT)
class GuiAssembler(playerInv: InventoryPlayer, tile: TileAssembler) extends GuiContainer(new ContainerAssembler(playerInv, tile)){
  private val TEXTURE: ResourceLocation = new ResourceLocation("exo", "texture/gui/guiAssembler.png");
  
  override def drawGuiContainerBackgroundLayer(f: Float, i: Int, j: Int){
    GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
    this.mc.getTextureManager().bindTexture(this.TEXTURE);
    val k: Int = (this.width - this.xSize) / 2;
    val l: Int = (this.height - this.ySize) / 2;
    this.drawTexturedModalRect(k, l, 0, 0, this.xSize, this.ySize);
  }
  
  override def drawGuiContainerForegroundLayer(x: Int, y: Int){
    
  }
}